pmd-core/etc/xslt/pmd-report-v2.xslt

```diff
@@ -0,0 +1,251 @@
+
+
+    <script src="https://cdn.jsdelivr.net/npm/vega@5.21.0"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-lite@5.2.0"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.20.2"></script>
+
+    <script>
+      let total = 0;
+      let p1 = 0 , p2 = 0, p3 = 0, p4 = 0, p5 = 0;
+      let bp =0, codeStyle = 0, design = 0, doc = 0, ep = 0, perf = 0, security = 0;
+    </script>
+   
+   </head>
+  <body>
+ <nav class="nav navbar navbar-expand-md navbar-dark bg-dark fixed-top">
+
+ 
+  <div class="container-fluid">
+    <a class="navbar-brand" href="#">PMD Report</a>
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+  
+  </div>
+</nav>
+  <div class='container' style='margin-top:100px;'>
+    <h5>Problems found ::  <small> Version: <xsl:value-of select="$version"/> @ <xsl:value-of select="$timestamp"/></small>
+     <span id="summary"></span>  </h5>
+    <table id='pmdTable' class='table table-hover table-striped'>
+    <thead>
+ 
+      <tr>
+        <th>File #</th>
+        <th>Issue #</th>
+        <th>File</th>
+        <th>Line</th>
+        <th>Priority</th>
+        <th>Rule</th>
+        <th>Problem</th>
+
+   
+      </tr>
+      </thead>
+      <xsl:for-each select="scan:file">
+      <xsl:variable name="filename" select="@name" />
+      <xsl:variable name="slno" select="position()" />
+         <xsl:for-each select="scan:violation">
+          <xsl:variable name="pos" select="position()" />
+           <xsl:variable name="currPos" select="position()" />
+          <tr>
+          <td><xsl:value-of select="$slno"/></td>
+          <td><xsl:value-of select="position()"/></td>
+          <td><xsl:value-of select="$filename"/> </td>
+          <td><xsl:value-of select="@beginline"/></td>
+          <td><xsl:value-of select="@priority"/></td>
+          <xsl:variable name="priority" select="@priority" />
+         <script> total++; </script>
+    <xsl:choose>
+      <xsl:when test="$priority = 1">
+        <script> p1++; </script>
+      </xsl:when>
+         <xsl:when test="$priority = 2">
+        <script> p2++; </script>
+      </xsl:when>
+
+      <xsl:when test="$priority = 3">
+        <script> p3++; </script>
+      </xsl:when>
+      <xsl:when test="$priority = 4">
+        <script> p4++; </script>
+      </xsl:when>
+      <xsl:when test="$priority = 5">
+        <script> p5++; </script>
+      </xsl:when>
+
+
+      <xsl:otherwise>
+        <xsl:text></xsl:text>
+      </xsl:otherwise>
+    </xsl:choose>
+
+
+   
+
+
+    <xsl:variable name="ruleset" select="@ruleset" />
+    <xsl:choose>
+      <xsl:when test="$ruleset = 'Best Practices'">
```

Comment:
Note: This mapping might not work, when custom rulesets are used.
Consider to add an "other" counter for all rulesets you couldn't map.